tests/unit/plugins/kubernetes/kubernetes_pod_executor_test.py

```diff
@@ -39,3 +49,32 @@ def test_run_updates_task_metadata(k8s_executor):
             ),
         },
     )
+
+
+def test_run(k8s_executor):
+    task_config = KubernetesTaskConfig(
+        name="fake_task_name",
+        uuid="fake_id",
+        image="fake_docker_image",
+        command="fake_command"
+    )
+    fake_container = V1Container(
+        image=task_config.image,
+        name=task_config.name,
+        command=[task_config.command]
+    )
+    fake_pod = V1Pod(
+        metadata=V1ObjectMeta(
+            name=task_config.pod_name,
+            namespace="task_processing_tests"
+        ),
+        spec=V1PodSpec(
+            restart_policy=task_config.restart_policy,
+            containers=[fake_container]
+        ),
+    )
+
+    assert k8s_executor.run(task_config) == task_config.pod_name
+    assert k8s_executor.api.create_namespaced_pod.call_args_list == [
```

Comment:
once we've moved to using kube_client in run, this line will probably look something like:
```suggestion
    assert k8s_executor.kube_client.core.create_namespaced_pod.call_args_list
```

Reply:
switched to using kube_client